realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -307,6 +307,27 @@ private boolean categorizeClassElements() {
                     }
                 }
 
+                if (variableElement.getAnnotation(Required.class) == null) {
+                    // The field doesn't have the @Required annotation
+                    if (!Utils.isPrimitiveType(variableElement) &&
+                            !Utils.isRealmList(variableElement)) {
+                        nullableElements.add(variableElement);
+                    }
+                } else {
+                    // The field has the @Required annotation
+                    if (Utils.isPrimitiveType(variableElement)) {
+                        Utils.error("@Required is not needed for primitive fields - got " + element);
+                    } else if (Utils.isRealmList(variableElement)) {
+                        Utils.error("@Required is not needed for RealmList fields - got " + element);
```

Comment:
`@Required` is not a possibility for a list, right? So the error message should be something like: "@Required is not valid for RealmList".